frb_codegen/src/config.rs

```diff
@@ -78,30 +79,109 @@ pub struct Opts {
+    let mut dart_output_paths = raw.dart_output.clone();
+    dart_output_paths.retain(|each_path| !each_path.trim().is_empty());
+    let dart_output_paths = dart_output_paths
+        .iter()
+        .map(|each_path| canon_path(each_path))
+        .collect::<Vec<_>>();
+    assert!(
+        dart_output_paths.len() == rust_input_paths.len(),
+        "dart output path(s) should have the same number of path(s) as rust input(s)"
+    );
+
+    // rust crate dir(s)
+    let rust_crate_dirs = raw.rust_crate_dir.unwrap_or_else(|| {
+        rust_input_paths
+            .iter()
+            .map(|each_rust_input_path| {
+                fallback_rust_crate_dir(each_rust_input_path)
+                    .unwrap_or_else(|_| panic!("{}", format_fail_to_guess_error("rust_crate_dir")))
+            })
+            .collect::<Vec<_>>()
     });
-    let c_output_path = raw
+    let rust_crate_dirs = rust_crate_dirs
+        .iter()
+        .map(|each_path| canon_path(each_path))
+        .collect::<Vec<_>>();
+    assert!(
+        rust_crate_dirs.len() == rust_input_paths.len(),
+        "rust crate dir(s) should have the same number of path(s) as rust input(s)"
+    );
+
+    // manifest path(s)
+    let manifest_paths = rust_crate_dirs
+        .iter()
+        .map(|each| {
+            let mut path = std::path::PathBuf::from_str(each).unwrap();
+            path.push("Cargo.toml");
+            path_to_string(path).unwrap()
+        })
+        .collect::<Vec<_>>();
+    assert!(
+        manifest_paths.len() == rust_input_paths.len(),
+        "manifest path(s) should have the same number of path(s) as rust input(s)"
+    );
+
+    // rust output path(s)
+    let rust_output_paths = match raw.rust_output {
+        Some(rust_outputs) => rust_outputs,
+        None => {
+            if rust_input_paths.len() == 1 {
+                vec![fallback_rust_output_path(&rust_input_paths[0])
+                    .unwrap_or_else(|_| panic!("{}", format_fail_to_guess_error("rust_output")))]
+            } else {
+                panic!("for more than 1 rust blocks, please specify each rust output path clearly with flag \"rust-output\"");
+            }
+        }
+    };
+    let rust_output_paths = rust_output_paths
+        .iter()
+        .map(|each_path| canon_path(each_path))
+        .collect::<Vec<_>>();
+    assert!(
+        rust_output_paths.len() == rust_input_paths.len(),
+        "rust output path(s) should have the same number of path(s) as rust input(s)"
+    );
+
+    // class name(s)
+    let class_names = match raw.class_name {
+        Some(class_names) => class_names,
+        None => {
+            if rust_input_paths.len() == 1 {
+                vec![fallback_class_name(&*rust_input_paths[0])
+                    .unwrap_or_else(|_| panic!("{}", format_fail_to_guess_error("class_name")))]
+            } else {
+                panic!("for more than 1 rust blocks, please specify each class name clearly with flag \"class-name\"");
```

Comment:
This duplicates with logic of rust_output. Shall we extract into method?

Reply:
Fixed in 55f2ef0aded946195114b076e329a631edab04b5, with new function `get_outputs_for_flag_requires_full_data`.